src/cow-react/modules/advancedOrders/containers/NumberOfParts/index.tsx

```diff
@@ -0,0 +1,32 @@
+import { WidgetField, WidgetLabel, WidgetContent } from '@cow/modules/advancedOrders/pure/WidgetField'
+import { Trans } from '@lingui/macro'
+import QuestionHelper from 'components/QuestionHelper'
+import * as styledEl from './styled'
+import { useAtomValue, useSetAtom } from 'jotai'
+import {
+  advancedOrdersSettingsAtom,
+  updateAdvancedOrdersSettingsAtom,
+} from '@cow/modules/advancedOrders/state/advancedOrdersSettingsAtom'
+
+export function NumberOfParts() {
+  const { numberOfParts } = useAtomValue(advancedOrdersSettingsAtom)
+
+  const updateSettingsState = useSetAtom(updateAdvancedOrdersSettingsAtom)
+
+  const onUserInput = (value: string) => {
+    updateSettingsState({ numberOfParts: Number(value) })
```

Comment:
What happens if `value` is not a number?

Reply:
I was not able to insert anything that's not an integer, so I guess this is covered in the component level

Reply:
it doesn't harm to do as you did in another PR and add a default or something, so you don't get a NaN

Reply:
+ What about decimals? I can't input `12.95` (we need it for slippage value)